source/lac/scalapack.cc

```diff
@@ -263,6 +263,131 @@ ScaLAPACKMatrix<NumberType>::operator=(const FullMatrix<NumberType> &matrix)
 
 
 
+template <typename NumberType>
+void
+ScaLAPACKMatrix<NumberType>::copy_from(const LAPACKFullMatrix<NumberType> &B,
+                                       const unsigned int                  r)
+{
+  if (n_rows * n_columns == 0)
+    return;
+
+  const unsigned int this_mpi_process(
+    Utilities::MPI::this_mpi_process(this->grid->mpi_communicator));
+
```

Comment:
Here and below you can add a check inside ‘#ifdef DEBUG’ that rank is the same everywhere. To that end u can get maximum over communicator via Utilities.